lib/kitchen/chapter_element.rb

```diff
@@ -44,6 +44,14 @@ def introduction_page
       pages('$.introduction').first
     end
 
+    # Returns true if the chapter has an introduction
+    #
+    # @return [Boolean]
+    #
+    def has_introduction?
+      introduction_page.present?
```

Comment:
memoize this so that future calls are free.  

```ruby
def has_introduction?
  @has_introduction ||= introduction_page.present?
end
```

Reply:
RIGHT